### DISCUSSION TITLE
How do debug Quarkus with so many exceptions being thrown?

### DISCUSSION THREAD

**Discussion by vladaman** [quarkusio/quarkus#27561] [Q&A] [answered]
I tried to debug an application in `mvn quarkus:dev` mode. Every time I reload the app I am getting MANY class not found exceptions thrown within the IDE. Specifically in `loadClass(String cn, boolean resolve)`

Is this normal? How do you guys debug in intellij without these class loading exceptions?

![image](https://user-images.githubusercontent.com/907206/187148406-88ba10d2-2b81-4203-bfe0-c1711b0a0df2.png)
![image](https://user-images.githubusercontent.com/907206/187148463-85821aca-d8b2-4509-a5eb-a9571b2ad0e3.png)
![image](https://user-images.githubusercontent.com/907206/187148584-f6227eb1-bbf9-43bb-a462-576d2b385165.png)
![image](https://user-images.githubusercontent.com/907206/187148646-23a60fdf-7a51-42e1-8034-11ceb527f154.png)

... and many more

**Comment by geoand** [comment, marked as answer]
Yes, it's absolutely normal because of the hierarchical nature of the ClassLoaders in dev mode.

Generally I have found that using breakpoints in ClassLoader code is difficult (even without Quarkus), because there are a ton of classes being loaded and because conditional breakpoints don't really work due to the recursiveness of the classloading mechanism
